src/app/CommandHandler.h

```diff
@@ -50,8 +53,34 @@ class CommandHandler : public Command
 
 private:
     friend class TestCommandInteraction;
+    friend class CommandHandlerHandleDeleter;
     CHIP_ERROR SendCommandResponse();
     CHIP_ERROR ProcessCommandDataElement(CommandDataElement::Parser & aCommandElement) override;
+
+    struct CommandHandlerHandleDeleter
+    {
+        void operator()(CommandHandler * apCommandHandler) const;
+    };
+
+    uint32_t mPendingWorkCount = 0;
+
+public:
+    using CommandHandlerAsyncHandle = std::unique_ptr<CommandHandler, CommandHandlerHandleDeleter>;
+    /**
+     * @brief Usually, the command handler do things very fast, and a synchronous handler is OK, however, some command handler might
+     * require some background work, thus we need to mark the current CommandHander as a async command handler.
+     * - Releasing the CommandHandlerAsyncHandle means the owner has finished the work and put necessary responses, and the response
+     * (status code or response command, determined by what has been put put in the response by cluster handler) will be sent by
+     * interaction model.
+     * - Releasing the CommandHandlerAsyncHandle before encode a response will result in failure during sending a response, and
```

Comment:
```suggestion
     * - Releasing the CommandHandlerAsyncHandle before encoding a response will result in failure during sending of the response, and
```
But will it?  Why would it result in that, exactly?  Especially if there are multiple invokes in the packet?

Reply:
Changed the statement to "might result in malformed response"